### DISCUSSION TITLE
The token contains no permissions, or permissions can not be understood.

### DISCUSSION THREAD

**Discussion by bhaskarvenna** [O365/python-o365#727] [Q&A]
Hi,
Can you please help me here. It shows authenticated, but i cant see any emails from my company inbox.

here is my error:
_**HTTPError: 401 Client Error: Unauthorized for url: https://graph.microsoft.com/v1.0/users/diannaraedle@deerislegroup.com/mailFolders/Inbox/messages?%24top=25 | Error Message: The token contains no permissions, or permissions can not be understood.**_

Here is my Code:
credentials = ('clientId','clientSecret')

account = Account(credentials, auth_flow_type='credentials', tenant_id='67cee803-626e-43b3-a4f0-45f90b5c7584')
if account.authenticate():
   print('Authenticated!')

mailbox = account.mailbox(abcd@mycompany.com)
print(dir(mailbox))

inbox = mailbox.inbox_folder()

for message in inbox.get_messages():
    print(message)

**Comment by alejcas** [comment]
Maybe because you need permissions for abcd@mycompany.com??
